EssentialCSharp.Web/Models/GuidelineListing.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System.Text.Json;
+
+namespace EssentialCSharp.Web.Models
+{
+    public record GuidelineListing(GuidelineType Type, string Guideline, int ChapterNumber, string? ChapterTitle, string SanitizedSubsection, string? ActualSubsection)
+    {
+        public static GuidelineType GetGuidelineType(IReadOnlyList<string> selectionString)
+        {
+            if (selectionString.Count < 2)
+            {
+                return GuidelineType.None;
+            }
+
+            if (selectionString[1].Equals("not", StringComparison.OrdinalIgnoreCase))
+            {
+                return GuidelineType.DoNot;
+            }
+
+            if (Enum.TryParse<GuidelineType>(selectionString[0], true, out GuidelineType result))
+            {
+                return result;
+            }
+            else
+            {
+                return GuidelineType.None;
+            }
+        }
+    }
+
+    public enum GuidelineType
+    {
+        None,
+        DoNot,
+        Avoid,
+        Consider,
+        Do
+    }
+
+    public static class GuidelineListingExtensions
+    {
+        private static readonly JsonSerializerOptions _Options = new() { WriteIndented = true };
+
+        public static List<GuidelineListing>? ReadGuidelineJsonFromInputDirectory(FileInfo guidelinesJsonFile, ILogger logger)
```

Comment:
This class is named liked an extension method class, but this is not an extension method.